### PR TITLE
Invalidate removed inventory items

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

File: Source/player.cpp [hunk 1]
```diff
@@ -1937,34 +1937,35 @@ bool Player::HasItem(int item, int *idx) const
...
 
-	iv--;
+	InvList[iv]._itype = ItemType::None;
+
 	_pNumInv--;
 
 	// If the item at the end of inventory array isn't the one we removed, we need to swap its position in the array with the removed item
 	if (_pNumInv > 0 && _pNumInv != iv) {
 		InvList[iv] = InvList[_pNumInv];
+		InvList[_pNumInv]._itype = ItemType::None;
```
Comment (mewmew) [inline Source/player.cpp:1954]: Looks good. I'd add a `memset(&InvList[_pNumInv], 0, sizeof(ItemStruct))` prior to this line, just to clear more garbage data from memory. And do the same on the above `._itype = ItemType::None;` line.

**Review by mewmew — COMMENTED** [review]
LGMT, with minor comments.

File: Source/player.cpp (hunk 1 above)
Comment (AJenbo) [reply Source/player.cpp:1954]: DevilutionX is C++: `InvList[_pNumInv] = {};` ;)

Comment (AJenbo) [reply Source/player.cpp:1954]: Going to opt out of zeroing just to keep thing simple for now.

**Pull request merged**